src/model/compatibility-check/check-field.ts

```diff
@@ -457,3 +463,56 @@ function checkCalcMutations(fieldToCheck: Field, baselineField: Field, context:
         return;
     }
 }
+
+/**
+ * Checks whether @root and @parent are specified exactly when they are specified in the baseline field
+ */
+function checkRootAndParentDirectives(
```

Comment:
Could be split up into two functions since the code does not really share any logic.

Reply:
yeah, but @parent and @root are so similar to each other that I thought having them together would be better. For example, if you would change something for `qparent`, you would most likely also want to change it for `@root`

Reply:
I am fine with it (was more of an optional suggestion)